### PR TITLE
docs/cli: update martinvonz.github.io URLs to jj-vcs.github.io

### PR DESCRIPTION
**Pull request by martinvonz**
We just migrated to the jj-vcs GitHub org, so we should point to the new GitHub pages.

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Review by PhilipMetzger — APPROVED** [review]

**Review by arxanas — APPROVED** [review]
comment (non-blocking): You're no doubt aware, but the root URL <https://jj-vcs.github.io> produces a 404, which we should try to fix at some point.

comment (non-blocking): It's weird that all these tests are failing 👀

**Comment by scott2000** [comment]
> comment (non-blocking): It's weird that all these tests are failing 👀

I think it's cause the README file added to conflict commits changed, so all conflicted commits have a different hash now.

File: lib/src/git_backend.rs [hunk 1]
```diff
@@ -1430,7 +1430,7 @@ Git or other tools that are unfamiliar with jj.
 
 The .jjconflict-* directories represent the different inputs to the conflict.
 For details, see
-https://martinvonz.github.io/jj/prerelease/git-compatibility/#format-mapping-details
+https://jj-vcs.github.io/jj/prerelease/git-compatibility/#format-mapping-details
```
Comment (scott2000) [inline lib/src/git_backend.rs:1433]: This file also contains a link to `https://github.com/martinvonz/jj` which should probably be updated.

Comment (martinvonz) [reply lib/src/git_backend.rs:1433]: Oh, good point! There are tons of those links. I'll fix all of them, including the ones in the changelog (it feels a bit weird to update the changelog, but I think it makes sense in this case).

**Comment by martinvonz** [comment]
> comment (non-blocking): You're no doubt aware, but the root URL https://jj-vcs.github.io produces a 404, which we should try to fix at some point.

I was not, but it makes sense that it fails. That's the user-/org-level GitHub Pages URL, so we'll need to set up a separate little repo for that (like https://github.com/martinvonz/martinvonz.github.io)

File: lib/src/git_backend.rs (hunk 1 above)
Comment (martinvonz) [reply lib/src/git_backend.rs:1433]: Done.

**Pull request merged**